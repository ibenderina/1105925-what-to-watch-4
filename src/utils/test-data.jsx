import {Film, parseComments, parseFilms, UserAccount} from "@api/adapter";
import NameSpace from "@reducer/name-space";
import {AuthorizationStatus} from "@reducer/user/user";
import {ALL_GENRES, TransferStates} from "@consts";

export const rawTestFilms = [
  {"name": `Once Upon a Time in America`, "poster_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg`, "preview_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg`, "background_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/ones_upon_a_time_in_america.jpg`, "background_color": `#CBAC79`, "description": `A former Prohibition-era Jewish gangster returns to the Lower East Side of Manhattan over thirty years later, where he once again must confront the ghosts and regrets of his old life.`, "rating": 9.9, "scores_count": 276395, "director": `Sergio Leone`, "starring": [`Robert De Niro`, `James Woods`, `Elizabeth McGovern`], "run_time": 229, "genre": `Crime`, "released": 1984, "id": 1, "is_favorite": false, "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`, "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`},
  {"name": `No Country for Old Men`, "poster_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/No_Country_for_Old_Men.jpg`, "preview_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/no-country-for-old-men.jpg`, "background_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/No_Country_for_Old_Men.jpg`, "background_color": `#BDAD8F`, "description": `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`, "rating": 4.1, "scores_count": 764976, "director": `Ethan Coen`, "starring": [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`], "run_time": 122, "genre": `Crime`, "released": 2007, "id": 2, "is_favorite": false, "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`, "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`},
  {"name": `A Star Is Born`, "poster_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/A_Star_Is_Born.jpg`, "preview_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/A_Star_Is_Born.jpg`, "background_image": `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/A_Star_is_Born.jpg`, "background_color": `#C4C0C0`, "description": `A musician helps a young singer find fame as age and alcoholism sendhis own career into a downward spiral.`, "rating": 3.9, "scores_count": 244484, "director": `Bradley Cooper`, "starring": [`Lady Gaga`, `Bradley Cooper`, `Sam Elliott`], "run_time": 136, "genre": `Drama`, "released": 2018, "id": 3, "is_favorite": false, "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`, "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`}
];

export const rawTestComments = [
  {"id": 1, "user": {"id": 11, "name": `Jack`}, "rating": 9.7, "comment": `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`, "date": `2020-06-21T10:54:44.818Z`},
  {"id": 2, "user": {"id": 10, "name": `Max`}, "rating": 4, "comment": `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`, "date": `2020-06-23T10:54:44.818Z`}
];

export const rawUserAccount = {"id": 1, "email": `test@te33st.tr`, "name": `test`, "avatar_url": `/wtw/static/avatar/5.jpg`};

export const testFilms = parseFilms(rawTestFilms);
export const testComments = parseComments(rawTestComments);
export const testUserAccount = UserAccount.parse(rawUserAccount);

export const testUserStore = {
  authorizationStatus: AuthorizationStatus.IN_PROGRESS,
  authorizationErrorMessage: ``,
  userAccount: testUserAccount
};

export const testUserEmptyStore = {
  authorizationStatus: AuthorizationStatus.IN_PROGRESS,
  authorizationErrorMessage: ``,
  userAccount: new UserAccount({}),
};

export const testFilmStore = {
  currentGenre: ALL_GENRES,
  genres: [...new Set(testFilms.map((f) => f.genre))],
  films: testFilms,
  promoFilm: testFilms[0],
  showedFilmsCount: 8
};

export const testEmptyFilmStore = {
  currentGenre: ALL_GENRES,
  genres: [],
  films: [],
  promoFilm: new Film({}),
  showedFilmsCount: 0,
};

export const testEmptyCommentsStore = {
  addStatusMessage: ``,
  addStatus: TransferStates.NEW,
};

export const testCommentsStore = {
  1: testComments,
  addStatusMessage: ``,
  addStatus: TransferStates.NEW,
};

export const testStore = {
  [NameSpace.USER]: testUserStore,
  [NameSpace.FILMS]: testFilmStore,
  [NameSpace.COMMENTS]: testCommentsStore,
};
