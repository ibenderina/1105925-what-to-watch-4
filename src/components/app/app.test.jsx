import App from "@components/app/app";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {ALL_GENRES} from "@consts";
import {testFilms} from "@utils/test-data";
import NameSpace from "@reducer/name-space";


const mockStore = configureStore();

const initialState = {
  currentGenre: ALL_GENRES,
  genres: [...new Set(testFilms.map((f) => f.genre))],
  films: testFilms,
  isMoreFilms: true,
  promoFilm: testFilms[0],
  showedFilmsCount: 8
};

it(`Render App`, () => {
  const store = mockStore({
    [NameSpace.FILMS]: initialState,
  });

  let tree;
  window.act(() => {
    tree = window.create(
        <Provider store={store}>
          <App/>
        </Provider>);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
