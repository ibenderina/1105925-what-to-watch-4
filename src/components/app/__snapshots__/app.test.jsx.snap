// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Gentlemen background"
        src="https://99px.ru/sstorage/53/2017/12/tmb_217309_894087.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Gentlemen poster"
            height="327"
            src="https://twizz.ru/wp-content/uploads/-000//1/animals-about-to-drop-album-photos-58aeb0ce59ff1__700.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Gentlemen
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy
            </span>
            <span
              className="movie-card__year"
            >
              One Million years AC
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Dramas
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Kids & Family
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Sci-Fi
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Thrillers
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Dead in a Week (Or Your Money Back)"
              height="175"
              src="https://pets.mail.ru/sharepic/68/10401.png"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Dead in a Week (Or Your Money Back)
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="The Gentlemen"
              height="175"
              src="https://twizz.ru/wp-content/uploads/-000//1/animals-about-to-drop-album-photos-58aeb0ce59ff1__700.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              The Gentlemen
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Lion"
              height="175"
              src="https://medialeaks.ru/wp-content/uploads/2019/08/glavnaya-1.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Lion
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Chappie"
              height="175"
              src="https://www.thepaws.net/wp-content/uploads/2018/10/funny-chihuahua.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Chappie
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Guns Akimbo"
              height="175"
              src="https://memepedia.ru/wp-content/uploads/2018/12/hamster.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Guns Akimbo
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Undisputed"
              height="175"
              src="https://stihi.ru/pics/2013/04/04/1852.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Undisputed
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="En man som heter Ove"
              height="175"
              src="https://techtoday.in.ua/wp-content/uploads/2018/01/GRUMPYCAT.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              En man som heter Ove
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Shining"
              height="175"
              src="https://cs7.pikabu.ru/post_img/2019/03/06/7/1551870696138123277.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Shining
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
  </section>,
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>,
]
`;
